### ISSUE TITLE
In-memory KeyStore on AWSIotKeystoreHelper from aws-sdk-android

### ISSUE THREAD

**Issue by leticiamendes2** [aws-amplify/aws-sdk-android#3089]
**State your question**
I would like to understand the usage and reasons for creating a temporary KeyStore when we call AWSIotKeystoreHelper.getIotKeystore. Also, I'd like to confirm that there is not in-memory KeyStore created when we use AWSIotMqttManager.connectUsingALPN.

**Which AWS Services are you utilizing?**
IoT

**Provide code snippets (if applicable)**

**Environment(please complete the following information):**
 - SDK Version: 2.51.0

**Device Information (please complete the following information):**
 - Device: 
 - Android Version: 8.1.0
 - Specific to simulators:

If you need help with understanding how to implement something in particular then we suggest that you first look into our [developer guide](https://docs.aws.amazon.com/aws-mobile/latest/developerguide/aws-mobile-android-and-iOS.html). You can also simplify your process of creating an application, as well as the associated backend setup by using the [Amplify CLI](https://aws-amplify.github.io/media/get_started).

**Comment by banji180** [comment]
Hello. Thank you for your questions. 

We have investigated your reported concern and we have determined that it does NOT represent a security issue. 

Q: I would like to understand the reasons for creating a temporary KeyStore when we call AWSIotKeystoreHelper.getIotKeystore. It doesn't seem necessary, and considering there will be a Private Key stored in that KeyStore it looks like a potential security issue.

A: In AWSIotKeystoreHelper.getIotKeystore(…), a temporary KeyStore is created in memory. The X509 certificate used to authenticate the device when connecting to AWS IoT and the private key for the certificate are added to this temporary KeyStore. AWSIotMqttManager.connectUsingALPN(…) uses this KeyStore to setup the certificate-based authentication on the socket for connecting to AWS IoT. Since the contents of the KeyStore are password protected but must be accessible in AWSIotMqttManager.connectUsingALPN(…), a known password (defined in AWSIotKeystoreHelper) is used for the KeyStore. As the KeyStore is protected with a known password, a temporary KeyStore is created in memory instead of using the KeyStore on device to prevent unauthorized access to the X509 certificate and private key. 

Q: Also, I'd like to confirm that there is not in-memory KeyStore created when we use AWSIotMqttManager.connectUsingALPN.
A: There is no in-memory KeyStore created in AWSIotMqttManager.connectUsingALPN(…).
